eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/OperationSourceProvider.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Generator.Primitives;
+using Azure.ResourceManager;
+using Microsoft.TypeSpec.Generator.ClientModel.Providers;
+using Microsoft.TypeSpec.Generator.Primitives;
+using Microsoft.TypeSpec.Generator.Providers;
+using Microsoft.TypeSpec.Generator.Statements;
+using System.ClientModel.Primitives;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using static Microsoft.TypeSpec.Generator.Snippets.Snippet;
+
+namespace Azure.Generator.Providers
+{
+    internal class OperationSourceProvider : TypeProvider
+    {
+        private string _resourceName;
+        private ResourceProvider _resource;
+        private ModelProvider _resourceData;
+        private CSharpType _operationSourceInterface;
+        private ModelSerializationExtensionsDefinition _modelSerializationExtensions;
+
+        private FieldProvider _clientField;
+
+        public OperationSourceProvider(string resourceName, ResourceProvider resource, ModelProvider resourceData, ModelSerializationExtensionsDefinition modelSerializationExtensions)
```

Comment:
Why does a ResourceProvider not know its name and data vs having to pass all 3 in separately?

Reply:
updated